OPHD/UI/StringTable.cpp

```diff
@@ -0,0 +1,221 @@
+#include "StringTable.h"
+#include "../FontManager.h"
+#include "../Constants/UiConstants.h"
+#include <NAS2D/Utility.h>
+#include <stdexcept>
+
+const NAS2D::Color StringTable::Cell::ColorEmpty = NAS2D::Color::NoAlpha;
+
+StringTable::Cell& StringTable::operator[](const CellCoordinate& coordinate)
+{
+	return cells[getCellIndex(coordinate)];
+}
+
+StringTable::Cell& StringTable::at(std::size_t column, std::size_t row)
+{
+	return cells[getCellIndex(CellCoordinate(column, row))];
+}
```

Comment:
I would like help with the design here. I was researching the operator[] and it seems a bit complicated to setup something like operator[][]. Also, I don't know that you can overload the operator[].

@DanRStevens, do you think it would be better to just have an overloaded 'at' function? This would be easiest to implement on my part.

Reply:
Hmm, seems you can't overload `operator[]` to take two comma separated parameters. I wouldn't advise trying to use `[][]` syntax.

----

I suppose you could just manually pack the parameters using `{}`, and then get rid of the `at` method. Calling code would look something like:
```cpp
	stringTable[{0, 0}].text = "Turns Completed:";
	stringTable[{1, 0}].text = std::to_string(mFactory->productionTurnsCompleted()) + " of " + std::to_string(mProductCost.turnsToBuild());

	stringTable[{0, 1}].text = "Common Metals:";
	stringTable[{1, 1}].text = std::to_string(mProductCost.commonMetals() * mProductCost.turnsToBuild());

	stringTable[{0, 2}].text = "Common Minerals:";
	stringTable[{1, 2}].text = std::to_string(mProductCost.commonMinerals() * mProductCost.turnsToBuild());

	stringTable[{0, 3}].text = "Rare Metals:";
	stringTable[{1, 3}].text = std::to_string(mProductCost.rareMetals() * mProductCost.turnsToBuild());

	stringTable[{0, 4}].text = "Rare Minerals:";
	stringTable[{1, 4}].text = std::to_string(mProductCost.rareMinerals() * mProductCost.turnsToBuild());
```

There is one complication internally with the line:
```cpp
		this->at(column, row).justification = justification;
```

This can be replaced with:
```cpp
		this->operator[]({column, row}).justification = justification;
```
Or with:
```cpp
		(*this)[{column, row}].justification = justification;
```